From 23c84fa1ec7e666cbb07fd5a97371e3afa6c3041 Mon Sep 17 00:00:00 2001
From: Bharath Ramesh <rameshb@vmware.com>
Date: Thu, 9 Oct 2014 13:09:49 -0700
Subject: [PATCH 3/7] Disable building Outlook and Exchange extensions

Reviewed by: akreher, arolett, swilsak
Review URL: https://reviewboard.eng.vmware.com/r/681099/

Conflicts:
	setup.py
---
 setup.py | 75 ++++++++++++++++++++++++++++++++++------------------------------
 1 file changed, 40 insertions(+), 35 deletions(-)

diff --git a/setup.py b/setup.py
index a695c1b..368960b 100644
--- a/setup.py
+++ b/setup.py
@@ -1982,41 +1982,46 @@ com_extensions += [
                         %(internet)s/PyIInternetProtocolSink.cpp    %(internet)s/PyIInternetSecurityManager.cpp
                     """ % dirs).split(),
                     depends=["%(internet)s/internet_pch.h" % dirs]),
-    WinExt_win32com('mapi', libraries="mapi32", pch_header="PythonCOM.h",
-                    sources=("""
-                        %(mapi)s/mapi.i                 %(mapi)s/mapi.cpp
-                        %(mapi)s/PyIABContainer.i       %(mapi)s/PyIABContainer.cpp
-                        %(mapi)s/PyIAddrBook.i          %(mapi)s/PyIAddrBook.cpp
-                        %(mapi)s/PyIAttach.i            %(mapi)s/PyIAttach.cpp
-                        %(mapi)s/PyIDistList.i          %(mapi)s/PyIDistList.cpp
-                        %(mapi)s/PyIMailUser.i          %(mapi)s/PyIMailUser.cpp
-                        %(mapi)s/PyIMAPIContainer.i     %(mapi)s/PyIMAPIContainer.cpp
-                        %(mapi)s/PyIMAPIFolder.i        %(mapi)s/PyIMAPIFolder.cpp
-                        %(mapi)s/PyIMAPIProp.i          %(mapi)s/PyIMAPIProp.cpp
-                        %(mapi)s/PyIMAPISession.i       %(mapi)s/PyIMAPISession.cpp
-                        %(mapi)s/PyIMAPIStatus.i        %(mapi)s/PyIMAPIStatus.cpp
-                        %(mapi)s/PyIMAPITable.i         %(mapi)s/PyIMAPITable.cpp
-                        %(mapi)s/PyIMessage.i           %(mapi)s/PyIMessage.cpp
-                        %(mapi)s/PyIMsgServiceAdmin.i   %(mapi)s/PyIMsgServiceAdmin.cpp
-                        %(mapi)s/PyIMsgStore.i          %(mapi)s/PyIMsgStore.cpp
-                        %(mapi)s/PyIProfAdmin.i         %(mapi)s/PyIProfAdmin.cpp
-                        %(mapi)s/PyIProfSect.i          %(mapi)s/PyIProfSect.cpp
-                        %(mapi)s/PyIConverterSession.i	%(mapi)s/PyIConverterSession.cpp
-                        %(mapi)s/PyIMAPIAdviseSink.cpp
-                        %(mapi)s/mapiutil.cpp
-                        %(mapi)s/mapiguids.cpp
-                        """ % dirs).split()),
-    WinExt_win32com_mapi('exchange', libraries="mapi32",
-                         sources=("""
-                                  %(mapi)s/exchange.i         %(mapi)s/exchange.cpp
-                                  %(mapi)s/PyIExchangeManageStore.i %(mapi)s/PyIExchangeManageStore.cpp
-                                  %(mapi)s/PyIExchangeManageStoreEx.i %(mapi)s/PyIExchangeManageStoreEx.cpp
-                                  %(mapi)s/mapiutil.cpp
-                                  """ % dirs).split()),
-    WinExt_win32com_mapi('exchdapi', libraries="mapi32",
-                         sources=("""
-                                  %(mapi)s/exchdapi.i         %(mapi)s/exchdapi.cpp
-                                  """ % dirs).split()),
+]
+if get_build_version() < 11.0: # check if python was built with vs2012
+    com_extensions += [
+        WinExt_win32com('mapi', libraries="mapi32", pch_header="PythonCOM.h",
+                        sources=("""
+                            %(mapi)s/mapi.i                 %(mapi)s/mapi.cpp
+                            %(mapi)s/PyIABContainer.i       %(mapi)s/PyIABContainer.cpp
+                            %(mapi)s/PyIAddrBook.i          %(mapi)s/PyIAddrBook.cpp
+                            %(mapi)s/PyIAttach.i            %(mapi)s/PyIAttach.cpp
+                            %(mapi)s/PyIDistList.i          %(mapi)s/PyIDistList.cpp
+                            %(mapi)s/PyIMailUser.i          %(mapi)s/PyIMailUser.cpp
+                            %(mapi)s/PyIMAPIContainer.i     %(mapi)s/PyIMAPIContainer.cpp
+                            %(mapi)s/PyIMAPIFolder.i        %(mapi)s/PyIMAPIFolder.cpp
+                            %(mapi)s/PyIMAPIProp.i          %(mapi)s/PyIMAPIProp.cpp
+                            %(mapi)s/PyIMAPISession.i       %(mapi)s/PyIMAPISession.cpp
+                            %(mapi)s/PyIMAPIStatus.i        %(mapi)s/PyIMAPIStatus.cpp
+                            %(mapi)s/PyIMAPITable.i         %(mapi)s/PyIMAPITable.cpp
+                            %(mapi)s/PyIMessage.i           %(mapi)s/PyIMessage.cpp
+                            %(mapi)s/PyIMsgServiceAdmin.i   %(mapi)s/PyIMsgServiceAdmin.cpp
+                            %(mapi)s/PyIMsgStore.i          %(mapi)s/PyIMsgStore.cpp
+                            %(mapi)s/PyIProfAdmin.i         %(mapi)s/PyIProfAdmin.cpp
+                            %(mapi)s/PyIProfSect.i          %(mapi)s/PyIProfSect.cpp
+                            %(mapi)s/PyIConverterSession.i	%(mapi)s/PyIConverterSession.cpp
+                            %(mapi)s/PyIMAPIAdviseSink.cpp
+                            %(mapi)s/mapiutil.cpp
+                            %(mapi)s/mapiguids.cpp
+                            """ % dirs).split()),
+        WinExt_win32com_mapi('exchange', libraries="mapi32",
+                             sources=("""
+                                      %(mapi)s/exchange.i         %(mapi)s/exchange.cpp
+                                      %(mapi)s/PyIExchangeManageStore.i %(mapi)s/PyIExchangeManageStore.cpp
+                                      %(mapi)s/PyIExchangeManageStoreEx.i %(mapi)s/PyIExchangeManageStoreEx.cpp
+                                      %(mapi)s/mapiutil.cpp
+                                      """ % dirs).split()),
+        WinExt_win32com_mapi('exchdapi', libraries="mapi32",
+                             sources=("""
+                                      %(mapi)s/exchdapi.i         %(mapi)s/exchdapi.cpp
+                                      """ % dirs).split()),
+    ]
+com_extensions += [
     WinExt_win32com('shell', libraries='shell32', pch_header="shell_pch.h",
                     windows_h_version = 0x600,
                     sources=("""
-- 
2.8.0.windows.1

